File: app/src/main/java/com/mithril/mobilegoldenleaf/database/converter/StatusConverter.java
package com.mithril.mobilegoldenleaf.database.converter;

import androidx.room.TypeConverter;

import com.mithril.mobilegoldenleaf.models.OrderStatus;


public class StatusConverter {

    @TypeConverter
    public String toString(OrderStatus orderStatus) {
        return orderStatus.name();
    }

    @TypeConverter
    public OrderStatus fromString(String orderStatus) {
        if (orderStatus != null) {
            return OrderStatus.valueOf(orderStatus);
        }
        return OrderStatus.PENDENTE;
    }
}

File: app/src/main/java/com/mithril/mobilegoldenleaf/database/converter/StatusConverter.kt
package com.mithril.mobilegoldenleaf.database.converter

import androidx.room.TypeConverter

import com.mithril.mobilegoldenleaf.models.OrderStatus


class StatusConverter {

    @TypeConverter
    fun toString(orderStatus: OrderStatus): String {
        return orderStatus.name
    }

    @TypeConverter
    fun fromString(orderStatus: String?): OrderStatus {
        return if (orderStatus != null) {
            OrderStatus.valueOf(orderStatus)
        } else OrderStatus.PENDENTE
    }
}
